lib.nix

```diff
@@ -1,41 +1,28 @@
 let
   # Allow overriding pinned nixpkgs for debugging purposes via plutus_pkgs
-  fetchNixPkgs = let try = builtins.tryEval <plutus_pkgs>;
+  iohkNix = import (
+    let try = builtins.tryEval <iohk_nix>;
     in if try.success
-    then builtins.trace "using host <plutus_pkgs>" try.value
-    else import ./fetch-nixpkgs.nix;
+    then builtins.trace "using host <iohk_nix>" try.value
+    else
+      let
+        spec = builtins.fromJSON (builtins.readFile ./iohk-nix.json);
+      in builtins.fetchTarball {
+        url = "${spec.url}/archive/${spec.rev}.tar.gz";
+        inherit (spec) sha256;
+      });
+  fetchNixpkgs = iohkNix.fetchNixpkgs ./nixpkgs-src.json;
+  pkgs = import fetchNixpkgs { config = {}; overlays = []; };
+  lib = pkgs.lib;
+  cleanSourceHaskell = iohkNix.cleanSourceHaskell pkgs;
+  getPackages = iohkNix.getPackages { inherit lib;};
 
-  maybeEnv = env: default:
-    let
-      result = builtins.getEnv env;
-    in if result != ""
-       then result
-       else default;
+  importPkgs = args: import fetchNixpkgs ({ overlays = [ iohkNix.jemallocOverlay ]; } // args);
```

Comment:
Should `haddockHydraOverlay` be used here since this was present before?

Reply:
no, haddock hydra overlay is applied to haskellPackages. We don't want to apply all the overlays to `importPkgs` which is the normal `pkgs` variable in default.nix.